alica_engine/src/engine/collections/RobotProperties.cpp

```diff
@@ -1,44 +1,27 @@
 #include <engine/AlicaEngine.h>
 #include <engine/PlanRepository.h>
 #include <engine/collections/RobotProperties.h>
+#include <engine/containers/AgentAnnouncement.h>
 #include <engine/model/CapValue.h>
 
 namespace alica
 {
 
-RobotProperties::RobotProperties() {}
-
-RobotProperties::RobotProperties(const AlicaEngine* engine, const std::string& name)
-{
-    readFromConfig(engine, name);
-}
-
-/**
- * Reads the default role as well as the capabilities and characteristics from the Globals.conf.
- */
-void RobotProperties::readFromConfig(const AlicaEngine* engine, const std::string& name)
+RobotProperties::RobotProperties(const AlicaEngine* engine, const AgentAnnouncement& aa)
+        : _defaultRole(aa.role.empty() ? "NOROLESPECIFIED" : aa.role)
```

Comment:
That won't work, AlicaRosCommunication would just replace it with whatever received from the network, which could be empty. It could be shifted there once we have a proper builder class for messages though.